File: src/main/java/org/bp/lab/simpleblog/service/PostService.java
package org.bp.lab.simpleblog.service;

import org.bp.lab.simpleblog.domain.Post;
import org.springframework.data.domain.Page;
import org.springframework.web.multipart.MultipartFile;

public interface PostService {
	
	Page<Post> findAll(int page, int size);
	Page<Post> findAllByBloggerId(Long id, int page, int size);
	Post findOne(Long id);
	Post save (Post post);
	Post saveWithFile(String post, MultipartFile file);
	void delete (Long id);

}

File: src/main/webapp/static/app/js/post/postService.js
var blog = angular.module('blog.services', []);

blog.service('postService', function($http){
	
	this.url = '/api/posts';
	
	this.getPage = function(page, size) {
		return $http.get(this.url, { params: { 'page': page, 'size': size} });
	};
	
	this.findAll = function(page, size) {
		return $http.get(this.url + '/admin', { params: { 'page': page, 'size': size }});
	};
	
	this.getPostsByBloggerId = function(id, page, size){
		return $http.get(this.url, {params: { 'id': id, 'page': page, 'size': size }});
	};

	this.getOne = function(id) {
		return $http.get(this.url + '/' + id);
	};

	this.save = function(post) {
		if(post.id){
			return $http.put(this.url + '/' + post.id, post);
		}else{
			return $http.post(this.url, post);
		}
	};
	
	this.saveWithFile = function(fd){
		return $http.post(this.url + '/file', fd, {
			transformRequest : angular.identity,
			headers : {
			'Content-Type' : undefined
			}
		});
	};

	this.delete = function(id) {
		return $http.delete(this.url + '/' + id);
	};
	
});